Plugins/nosWebRTC/Source/WebSocketClient.cpp

```diff
@@ -123,39 +144,52 @@ void nosWebSocketClient::StartWebSocket()
 	Info.gid = -1;
 	Info.uid = -1;
 	Info.user = this;
+	Info.timeout_secs = 30;
 
 	Info.options |= LWS_SERVER_OPTION_DISABLE_IPV6;
+	Info.options |= LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT;
+
+	// Set the paths to your self-signed certificate and private key files
+	std::string certPath = std::string(nosEngine.Module->RootFolderPath) + "/cert.pem";
+	std::string keyPath = std::string(nosEngine.Module->RootFolderPath) + "/key.pem";
+	Info.client_ssl_cert_filepath = certPath.c_str();
+	Info.client_ssl_private_key_filepath = keyPath.c_str();
+
 	pContext = lws_create_context(&Info);
 	if (!pContext)
-		throw std::exception("Port not found in the addres!");
+		throw std::exception("Port not found in the address!");
 
 	if (!path.starts_with('/')) {
 		path = '/' + path;
 	}
-	struct lws_client_connect_info connect_info = {};
 
+	struct lws_client_connect_info connect_info = {};
 	connect_info.context = pContext;
-	connect_info.address = serverAddres.c_str(); // The address of the WebSocket server
-	connect_info.port = port; // Port of the WebSocket server
-	connect_info.path = path.c_str(); // The path of the WebSocket resource
-	connect_info.host = serverAddres.c_str(); // Set the hostname
-	connect_info.origin = "Nodos"; // Origin might be used by server for security checks to prevent possible attacks
-	connect_info.protocol = Protocols[0].name; // The first protocol we've defined
-
+	connect_info.address = serverAddres.c_str();
+	connect_info.port = port;
+	connect_info.path = path.c_str();
+	connect_info.host = serverAddres.c_str();
+	connect_info.origin = "Nodos";
+	connect_info.protocol = Protocols[0].name;
+
+	connect_info.ssl_connection = LCCSCF_USE_SSL | LCCSCF_ALLOW_SELFSIGNED | LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK | LCCSCF_ALLOW_EXPIRED | LCCSCF_ALLOW_INSECURE;
 	pWSI = lws_client_connect_via_info(&connect_info);
 	if (!pWSI)
-		throw std::exception("LWS connection failed!");
-
+	{
+		lws_context_destroy(pContext); // Clean up
+		throw std::runtime_error("Failed to connect to server");
+	}
 }
 
 std::tuple<std::string, int, std::string> nosWebSocketClient::ResolveAddres(std::string fullAddress)
 {
 	std::string server;
 	int port;
 	std::string path;
-	if ("ws://" == fullAddress.substr(0, 5)) {
+	if ("ws://" == fullAddress.substr(0, 5))
 		fullAddress = fullAddress.substr(5, fullAddress.size() - 5);
-	}
+	else if ("wss://" == fullAddress.substr(0, 6))
+		fullAddress = fullAddress.substr(6, fullAddress.size() - 6);
```

Comment:
This resolve address should return a struct like: 
```cpp
struct ParsedAddressInfo {
    std::string Scheme; // wss, http etc.
    std::string Address;
    std::string Path;
    int Port;
}
```

This can be changed according to the need. If the user did not provide a scheme, we could either set error or a default scheme.